inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/DorisExtractNode.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonTypeName;
+import org.apache.inlong.sort.protocol.FieldInfo;
+import org.apache.inlong.sort.protocol.constant.DorisConstant;
+import org.apache.inlong.sort.protocol.node.ExtractNode;
+import org.apache.inlong.sort.protocol.transformation.WatermarkField;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * doris extract node using doris flink-doris-connector-1.13.5_2.12
+ */
+@EqualsAndHashCode(callSuper = true)
+@JsonTypeName("dorisExtract")
+@Data
+public class DorisExtractNode extends ExtractNode implements Serializable {
+
+    private static final long serialVersionUID = -1369223293553991653L;
+
+    @JsonProperty("fenodes")
+    @Nonnull
+    private String feNodes;
+
+    @JsonProperty("username")
+    @Nonnull
+    private String userName;
+
+    @JsonProperty("password")
+    @Nonnull
+    private String password;
+
+    @JsonProperty("table.identifier")
+    @Nonnull
+    private String tableIdentifier;
+
+    @JsonCreator
+    public DorisExtractNode(@JsonProperty("id") String id,
+                            @JsonProperty("name") String name,
+                            @JsonProperty("fields") List<FieldInfo> fields,
+                            @Nullable @JsonProperty("watermarkField") WatermarkField waterMarkField,
+                            @JsonProperty("properties") Map<String, String> properties,
+                            @JsonProperty("fenodes") @Nonnull String feNodes,
+                            @JsonProperty("username") String userName,
+                            @JsonProperty("password") String password,
+                            @JsonProperty("table.identifier") String tableIdentifier) {
+        super(id, name, fields, waterMarkField, properties);
+        this.feNodes = Preconditions.checkNotNull(feNodes, "fenodes is null");
+        this.userName = Preconditions.checkNotNull(userName, "username is null");
+        this.password = Preconditions.checkNotNull(password, "password is null");
+        this.tableIdentifier = Preconditions.checkNotNull(tableIdentifier, "table.identifier is null");
+    }
+
+    @Override
+    public Map<String, String> tableOptions() {
+        Map<String, String> options = super.tableOptions();
+        options.put(DorisConstant.CONNECTOR, "doris");
+        options.put(DorisConstant.FE_NODES, feNodes);
+        options.put(DorisConstant.USERNAME, userName);
+        options.put(DorisConstant.PASSWORD, password);
+        options.put(DorisConstant.TABLE_IDENTIFIER, tableIdentifier);
+
+        return options;
+    }
+
+    @Override
+    public String genTableName() {
+        return String.format("table_%s", super.getId());
+    }
+
+    @Override
+    public List<FieldInfo> getPartitionFields() {
```

Comment:
It is recommend to remove it because of no change between it and the supper class.

Reply:
ok, i will fix it